index.js

```diff
@@ -349,10 +349,12 @@ class Raspi extends EventEmitter {
 
   digitalWrite(pin, value) {
     const pinInstance = this[getPinInstance](this.normalize(pin));
-    if (pinInstance.mode != OUTPUT_MODE) {
+    if (pinInstance.mode === INPUT_MODE && value === HIGH) {
+      pinInstance.peripheral = new DigitalInput({ pin: this.normalize(pin), pullResistor: PULL_DOWN });
+    } else if (pinInstance.mode != OUTPUT_MODE) {
       this.pinMode(pin, OUTPUT_MODE);
     }
-    if (value != pinInstance.previousWrittenValue) {
+    if (pinInstance.mode === OUTPUT_MODE && value != pinInstance.previousWrittenValue) {
```

Comment:
When `digitalWrite(pin, HIGH)` is called when the pin is in INPUT_MODE then the pull resistor is set but the pin remains as an INPUT. We need to guard against trying to write to the pin because this isn't possible for input pins.

I could instead return out of the function after setting the pull resistor on line 353?